### PR TITLE
[WASI] Not to build WASI-NN and its test except the WASI-NN flag.

### PR DESCRIPTION
**Pull request by q82419**
Signed-off-by: YiYing He <yiying@secondstate.io>

### CODE REVIEW

File: plugins/wasi_nn/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # SPDX-License-Identifier: Apache-2.0
 # SPDX-FileCopyrightText: 2019-2022 Second State INC
 
+# add backends building flags
+message(STATUS "Build OpenVINO backend for WASI-NN")
+add_definitions(-DWASMEDGE_WASINN_BUILD_OPENVINO)
```
Comment (hydai) [inline plugins/wasi_nn/CMakeLists.txt:6]: Use `WASMEDGE_WASINN_BACKEND="OpenVINO;..."` is better.

**Review by hydai — CHANGES_REQUESTED** [review]

File: plugins/wasi_nn/CMakeLists.txt (hunk 1 above)
Comment (q82419) [reply plugins/wasi_nn/CMakeLists.txt]: Done.

File: plugins/wasi_nn/CMakeLists.txt [hunk 2]
```diff
@@ -32,10 +32,7 @@ target_link_libraries(wasmedgePluginWasiNN
   wasmedgeSystem
 )
 
-# Only Linux systems support wasi-nn with openVINO now.
-if(CMAKE_SYSTEM_NAME MATCHES "Linux" AND WASMEDGE_WASINN_BUILD_OPENVINO)
-  install(TARGETS wasmedgePluginWasiNN DESTINATION ${CMAKE_INSTALL_LIBDIR}/wasmedge)
-endif()
+install(TARGETS wasmedgePluginWasiNN DESTINATION ${CMAKE_INSTALL_LIBDIR}/wasmedge)
```
Comment (hydai) [inline plugins/wasi_nn/CMakeLists.txt:35]: Looks like the wasi-nn plugin must be built and installed?

**Review by hydai — CHANGES_REQUESTED** [review]

File: utils/docker/build-wasinn-ubuntu-openvino.sh [hunk 3]
```diff
@@ -2,6 +2,16 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 # SPDX-FileCopyrightText: 2019-2022 Second State INC
 
+if [[ ! -v "${OPENVINO_VERSION}" ]]; then
```
Comment (hydai) [inline utils/docker/build-wasinn-ubuntu-openvino.sh:5]: I am not sure if we do want the wasinn/openvino related scripts in the `docker` folder.
Maybe putting these in the script folder is better?

**Review by hydai — CHANGES_REQUESTED** [review]

File: plugins/wasi_nn/CMakeLists.txt (hunk 2 above)
Comment (hydai) [reply plugins/wasi_nn/CMakeLists.txt:35]: Got it.

File: utils/docker/build-wasinn-ubuntu-openvino.sh (hunk 3 above)
Comment (q82419) [reply utils/docker/build-wasinn-ubuntu-openvino.sh:5]: The `build.sh` and `build-manylinux.sh` are also in the `docker` folder.
Maybe we can move them into `utils/build-scripts` together in the other PR?

Comment (hydai) [reply utils/docker/build-wasinn-ubuntu-openvino.sh:5]: Please notice that the `build.sh` and `build-manylinux.sh` are highly related to their corresponding dockerfiles, that's why we put them here. However, the OpenVINO/WASI-NN are not related to these dockerfiles.

**Review by hydai — CHANGES_REQUESTED** [review]

File: utils/wasi-nn/build-wasinn-ubuntu-openvino.sh [hunk 4]
```diff
@@ -2,6 +2,16 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 # SPDX-FileCopyrightText: 2019-2022 Second State INC
```
[the remark this answers is not in the document]
Comment (q82419) [reply utils/wasi-nn/build-wasinn-ubuntu-openvino.sh:4]: The book shows the commands directly. These scripts are used in CI.

File: utils/docker/build-wasinn-ubuntu-openvino.sh (hunk 3 above)
Comment (q82419) [reply utils/docker/build-wasinn-ubuntu-openvino.sh:5]: Done. Moved into `utils/wasi-nn`.

**Review by hydai — APPROVED** [review]

**Pull request merged**